src/Sylius/Bundle/CoreBundle/Resources/config/validation/PromotionRule.xml

```diff
@@ -13,89 +13,90 @@
 
 <constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping http://symfony.com/schema/dic/services/constraint-mapping-1.0.xsd">
     <class name="Sylius\Component\Promotion\Model\PromotionRule">
-        <constraint name="Sylius\Bundle\ApiBundle\Validator\Constraints\PromotionRuleGroup">
-            <option name="groups">sylius_api</option>
-        </constraint>
-
         <property name="configuration">
             <constraint name="Collection">
-                <option name="groups">cart_quantity</option>
+                <option name="allowExtraFields">true</option>
```

Comment:
Just to clarify, these are only needed because of a bug in our forms used in the UI.
In those, changing the type does not remove the old configuration and both, new and old, get sent with the request. It'll be addressed in a separate PR since we're not sure of the scope atm

Reply:
exactly